extensions/default/src/utils/getDirectURL.ts

```diff
@@ -49,10 +49,11 @@ const getDirectURL = (config, params) => {
         const options = {
           mediaType: defaultType,
         };
-        return value.retrieveBulkData(options).then(arr => {
+        const directRetrieveURL = await value.retrieveBulkData(options).then(arr => {
```

Comment:
Please remove this code rather than returning it like this.  The call to retrieveBulkData was a bad idea (of mine unfortunately).  There should be a separate method retrieveDataUrl instead containing the retrieve object logic.  This method should be fully synchronous all the time (no async AND no call returning a promise)

IF the bulkdata uri can directly return the URL without needing to include authorization, then it should return the URL.

Reply:
Sorry i'm lost with this PR as I don't catch the fetching logic with all these endpoint,

However for this poins I don't see how we can guess if the authorization header is needed or not, if an authorization system is in the dicom web it is likely to affect all routes including direct urls endpoint, otherwise it sounds like a security breach.

is it possible / expected to have a public endpoint on a dicom webserver with an authorization policy ?